### PR TITLE
Changed handling of content data to have new field: index time.

### PR DESCRIPTION
**Pull request by yarondbb**
Loading apllication data base from conent data is not supported yet in this commit

### CODE REVIEW

File: lib/content_data/content_data.rb [hunk 1]
```diff
@@ -70,8 +70,8 @@ def clone_contents_info
         instances_db_enum = instances_db.each_key
         loop {
           location =  instances_db_enum.next rescue break
-          instance_mtime = instances_db[location]
-          instances_db_cloned[[location[0].clone,location[1].clone]]=instance_mtime
+          inst_mod_times = instances_db[location]
+          instances_db_cloned[[location[0].clone,location[1].clone]] = inst_mod_times
```
Comment (genadyp) [inline lib/content_data/content_data.rb]: Suggest to add comment why we are using clone here

**Comment by genadyp** [comment]
@yarondbb 
[Fix for the time_modification_test](https://github.com/bbfsdev/bbfs/commit/2c408c1edb55d57d9a6e04cb60688785a1ad0b8e)
while clone contents_info object, then instances arrays should be also cloned

Actually it fixes all the failures of the rake:

```
f1 bbfs]$ rake
...
Finished tests in 3.149760s, 5.7147 tests/s, 30.4785 assertions/s.
18 tests, 96 assertions, 0 failures, 0 errors, 0 skips
....
Finished in 0.8018 seconds
42 examples, 0 failures
```

**Comment by yarondbb** [comment]
I see other errors during rake:

Failures:

  1) Index validation with shallow check succeeds when files exist and their attributes the same as in the index
     Failure/Error: @index.validate().should be_true
       expected: true value
            got: false
     # ./spec/content_data/validations_spec.rb:37:in `block (3 levels) in module:Spec'

  2) Index validation validation params :failed param returns index of absent or invalid instances
     Failure/Error: failed.contents_size.should eq(2)

```
   expected: 2
        got: 3

   (compared using ==)
 # ./spec/content_data/validations_spec.rb:102:in `block (3 levels) in <module:Spec>'
```

  3) Index validation of index from other file system succeeds when all remote contents has valid local instance
     Failure/Error: IndexValidations.validate_remote_index(@remote_index, @index).should be_true
       expected: true value
            got: false
     # ./spec/validations/index_validations_spec.rb:36:in `block (3 levels) in module:Spec'

  4) Index validation of index from other file system :failed param returns index of absent contents or contents without valid instances
     Failure/Error: failed.contents_size.should eq(2)

```
   expected: 2
        got: 3

   (compared using ==)
 # ./spec/validations/index_validations_spec.rb:65:in `block (3 levels) in <module:Spec>'
```

Finished in 0.72646 seconds
42 examples, 4 failures

**Comment by yarondbb** [comment]
Fxied in last commit: https://github.com/bbfsdev/bbfs/commit/6f0a64925c62273acfbada96b954f04ed254645d

Can we close this ?

**Pull request merged**